app/Model/MispObject.php

```diff
@@ -725,4 +725,147 @@ public function removeOrphanedObjects()
+                    'event_id' => $event_id,
+                    'object_id' => 0
+                ),
+            )
+        );
+        // $attribute_types = array_keys(Hash::combine($attributes, '{n}.Attribute.type'));
+        if (empty($attributes)) {
+            return array('templates' => array(), 'types' => array());
+        }
+        $attribute_types = array();
+        foreach ($attributes as $i => $attribute) {
+            $attribute_types[$attribute['Attribute']['type']] = 1;
+            $attributes[$i]['Attribute']['object_relation'] = $attribute['Attribute']['type'];
+        }
+        $attribute_types = array_keys($attribute_types);
+        $db = $this->getDataSource();
+        // # TEST 1
+        // array_walk($attribute_types, function(&$value) { $value = '"' . $value . '"'; });
+        // $potential_templates = $db->fetchAll(
+        //     'SELECT object_templates.id, object_templates.name, count(object_template_elements.type) as type_count FROM object_templates '
+        //         .'RIGHT JOIN object_template_elements ON object_templates.id = object_template_elements.object_template_id '
+        //         .'WHERE object_templates.active=1 AND object_template_elements.type IN (' . implode(',', $attribute_types) . ') '
+        //         .'GROUP BY object_templates.name ORDER BY type_count DESC;'
+        // );
+        // $potential_template_ids = Hash::extract($potential_templates, '{n}.object_templates.id');
+
+        // TEST 2
+        // $potential_templates = $this->ObjectTemplate->find('all', array(
+        //     'recursive' => -1,
+        //     'fields' => array(
+        //         'ObjectTemplate.id',
+        //         'ObjectTemplate.name',
+        //         'COUNT(ObjectTemplateElement.type) as type_count'
+        //     ),
+        //     'conditions' => array(
+        //         'ObjectTemplate.active' => true,
+        //     ),
+        //     'contain' => array(
+        //         'ObjectTemplateElement' => array(
+        //             'fields' => array('ObjectTemplateElement.object_relation', 'ObjectTemplateElement.type'),
+        //             'conditions' => array('ObjectTemplateElement.type' => $attribute_types)
+        //         )
+        //     ),
+        //     'group' => 'ObjectTemplate.name',
+        //     'order' => 'type_count DESC'
+        // ));
+
+        $potential_templates = $this->ObjectTemplate->find('all', array(
+            'recursive' => -1,
+            'fields' => array(
+                'ObjectTemplate.id',
+                'ObjectTemplate.name',
+                'COUNT(ObjectTemplateElement.type) as type_count'
+            ),
+            'conditions' => array(
+                'ObjectTemplate.active' => true,
+                'ObjectTemplateElement.type' => $attribute_types
+            ),
+            'joins' => array(
+                array(
+                    'table' => $db->fullTableName($this->ObjectTemplate->ObjectTemplateElement),
```

Comment:
Can just replace this with

'table' => 'object_template_elements'

Reply:
Though it was good practice.. ;)